spec/integration/fluent_logger_rails/tagged_hash_formatter_spec.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+require 'spec_helper'
+
+RSpec.describe FluentLoggerRails::TaggedHashFormatter do
+  around(:each) { |example| Time.use_zone('Pacific Time (US & Canada)') { example.run } }
+  before { Timecop.freeze('2019-01-08 14:51:39.701-0800') }
+  after { Timecop.return }
+
+  subject(:formatter) { described_class.new }
+
+  describe '#datetime_format' do
+    before do
+      formatter.datetime_format = '%Y-%m-%d'
+    end
+
+    it 'formats the date' do
+      expect(formatter.call('debug', DateTime.now, nil, 'hi')[:timestamp]).to eq '2019-01-08'
+    end
+  end
+
+  describe '#parent_key' do
+    before do
+      formatter.parent_key = :payload
+    end
+
+    it 'formats the date' do
+      expect(formatter.call(0, DateTime.now, nil, 'hi')).to eq(
+        payload: {
+          severity: 'DEBUG',
+          message: 'hi',
+          timestamp: '2019-01-08T14:51:39.700999 '
```

Comment:
Where does this trailing whitespace come from? Is it really necessary? 🤔

Reply:
Oh yeah, I noticed that before, good question. It looks like it's coming from the default `Logger::Formatter#format_datetime` function:
<img width="647" alt="Screen Shot 2020-08-18 at 9 07 01 AM" src="https://user-images.githubusercontent.com/245096/90537473-38749400-e132-11ea-8927-ac502b455bf5.png">

It looks like it's probably being "helpful" and putting a space in there so you get separation between the date and the PID. `I, [2019-01-08T14:51:39.700999 #1234] ....`. Little odd for sure, I can add a default format that works better with a hash/json payload aka no trailing space.